### DISCUSSION TITLE
detail.php : Different behavior between Nomintaim 3.3.0 and 4.20

### DISCUSSION THREAD

**Discussion by HubertNo** [osm-search/Nominatim#3566] [Q&A]
I don't know exactly which version this change appeared in, but the information returned by detail.php differs.

## What did you search for?

I make a request with the parameter addressdetails=1
for example : 
https://nominatim.openstreetmap.org/details.php?osmtype=R&osmid=59874&addressdetails=1&format=json&prety=1

## What result did you get?

I get in the address array at the end (OK designates an id or a correct value) :
index 4 : “France métropolitaine”, place_id OK, osm_id OK, admin_level 4
index 5 : “fr”, type “contry_code”, ...
index 6 : “France”, type “country”, place_id NULL, osm_id NULL, admin_level NULL

## What result did you expect?

But previously, on version 3.3.0, I got :
index 6: “France”, type “country”, place_id OK, osm_id OK, admin_level OK

What is the reason for this change? I haven't found any trace of it in the change logs.
Is it a regression or a change in behavior?

I'm attaching two screenshots, one with the current version of nominatim, the second with our old server version 3.3.0.

Thanks for your answers. 
Best regards.

<img width="1032" alt="Nominatim openstreemap nantes adress" src="https://github.com/user-attachments/assets/e5072edc-47f6-43b9-8537-6e774027c448">
<img width="939" alt="Nominatim3 3 0 heredis nantes address" src="https://github.com/user-attachments/assets/19db75a2-7e68-4f8e-92db-2d5b666bced5">

**Comment by mtmail** [comment]
The change happened around 2 years ago (Nominatim 4.1) and was deliberate. The country lookup moved to a different table, or rather the extra lookup of country was no longer needed. Basically every place in the database (`placex` table) already has a country_code value and Nominatim will only lookup the name from the `country_names` table, not all detail of the country. That's still the same in the current stable version Nominatim 4.5. (relevant change https://github.com/osm-search/Nominatim/pull/2614)

**Comment by HubertNo** [comment]
Hi,
Thanks for your answer. 
I now know which direction to go.
:-)
Regards.
